numpy/_core/_add_newdocs.py

```diff
@@ -4112,11 +4112,12 @@
     """
     a.partition(kth, axis=-1, kind='introselect', order=None)
 
-    Rearranges the elements in the array in such a way that the value of the
-    element in kth position is in the position it would be in a sorted array.
-    All elements smaller than the kth element are moved before this element and
-    all equal or greater are moved behind it. The ordering of the elements in
-    the two partitions is undefined.
+    Partially sort the array so that in the resultant array all elements
+    before the k-th element are less than that element, and all elements
+    after the k-th element are greater than or equal to it. Additionally,
+    the k-th element in the resulting array is in the position it would
+    occupy in a fully sorted array. The order of the elements in the two
+    partitions on either side of the k-th element is undefined.
```

Comment:
I feel like the original text is more readable except for the use of `before` and `behind`, which I find confusing and ambiguous, so I'd suggest keeping tha text but coming up with new wording there. Just my take.

Reply:
Agreed, "additionally" isn't additional at all in this order.  The old order was better.  The only thing I think may be a bit of an improvement is the "Partially sort" just to drop the word "sort" somewhere.